src/components/site-header.marko

```diff
@@ -170,4 +170,52 @@ window.addEventListener("keyup", (e) => {
         closeNavigation();
     }
 });
-</script>
\ No newline at end of file
+
+// Legacy link support: For the hash-based links dating before Jan 2025 when we transitioned to a multi-page site
+// For example, redirects from /#menu-button-radio to /component/menu-button#menu-button-radio
+document.addEventListener("DOMContentLoaded", function() {
```

Comment:
Note: This waits for `DOMContentLoaded` so it can verify that the currently referenced `#hash` link doesn't already exist as a link to the current page (to prevent accidental redirects). Note that this code short circuits on component/guide pages anyway (see below); but this check is just to be safe. Of course, if we ran synchronously here, the DOM parser wouldn't have had time to load elements on the page so a query select would fail (thus the need to wait a little).